# Tcruzi-GenomeCentromeres

Genome-wide identification and 3D characterization of centromeres in Trypanosoma cruzi using Hi-C and ChIP-seq, providing an improved Dm28c reference genome and revealing centromere-associated compartment boundaries and spatial clustering.


This repository contains the bioinformatics pipelines and scripts developed for the analysis of the *Trypanosoma cruzi* Dm28c strain, improving genome assembly and investigate centromere organization through KKT-enriched peak analysis.

## ðŸ§¬ Background & Abstract

*Trypanosoma cruzi*, the agent of Chagas disease, has a highly repetitive genome that complicates assembly. This pipeline was designed to support the generation of a high-quality reference genome for the **Dm28c strain**. By utilizing ChIP-seq data for kinetochore proteins (**KKT2** and **KKT3**) and Hi-C data, we identified  33 centromeric peaks across 18 scaffolds and 9 contigs.

The scripts in this repository allow for:
* **Structural Boundary Analysis**: Identifying centromeres as structural boundaries and their role in 3D spatial clustering.
* **Genomic Compartmentalization**: Analyzing the boundaries between conserved genes and virulence-factor multigene families.
* **CCN Variation**: Visualizing chromosome-wide copy number variations to ensure assembly accuracy and biological insight.

## ðŸ“„ Citation & Pre-print

For more details on the methodology and biological findings, please refer to our pre-print:

- NatÃ¡lia Karla Bellini, Pedro Gabriel Nachtigall, Pedro Leonardo Carvalho de Lima, et al. Genome-wide identification and 3D characterization of centromeres in Trypanosoma cruzi. bioRxiv 2025.09.26.678825; doi:https://doi.org/10.1101/2025.09.26.678825
